### PR TITLE
✨Consent: Allow using metaTag to block AMP components by tagName

### PR DESCRIPTION
**Pull request by zhouyx**
Partial for #20253 

I'll make the validator change in a separate PR.

### CODE REVIEW

File: src/consent.js [hunk 1]
```diff
@@ -73,3 +74,31 @@ export function getConsentPolicyInfo(element, policyId) {
...
+  if (!content) {
+    return false;
+  }
+
+  if (isArray(content)) {
+    content = content.join(',');
+  }
+
+  // Handles whitespace
+  content = content.replace(/\s/g, '_').split(',');
```
Comment (lannka) [inline src/consent.js]: you can do `toLowerCase` here to make the `if` simpler.

File: test/unit/test-consent.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      meta.setAttribute('name', 'amp-consent-blocking');
+      meta.setAttribute('content', 'amp-this');
+      head.appendChild(meta);
+      const element = doc.createElement('amp-this');
+      element.getAmpDoc = () => {return env.ampdoc;};
+      doc.body.appendChild(element);
+      expect(shouldBlockOnConsentByMeta(element)).to.be.true;
+    });
+
+    it.only('handles white space', () => {
```
Comment (lannka) [inline test/unit/test-consent.js]: remove `only`

File: src/consent.js [hunk 3]
```diff
@@ -73,3 +74,31 @@ export function getConsentPolicyInfo(element, policyId) {
...
+export function shouldBlockOnConsentByMeta(element) {
+  const ampdoc = element.getAmpDoc();
+  let content =
+      Services.documentInfoForDoc(ampdoc).metaTags['amp-consent-blocking'];
+
+  if (!content) {
+    return false;
+  }
+
+  if (isArray(content)) {
```
Comment (zhouyx) [inline src/consent.js]: Looks like I can apply the uniqueness check to `<meta amp-consent-blocking>`, so should be safe removing the check here. Let me know if you also prefer that. 

In addition, any idea on the naming? Thank you

File: src/consent.js [hunk 4]
```diff
@@ -73,3 +74,34 @@ export function getConsentPolicyInfo(element, policyId) {
...
+  let content =
+      Services.documentInfoForDoc(ampdoc).metaTags['amp-consent-blocking'];
+
+  if (!content) {
+    return false;
+  }
+
+  // validator enforce uniqueness of <meta name='amp-consent-blocking'>
+  // content will not be an array.
+  if (typeof content !== 'string') {
```
[the remark this answers is not in the document]
Comment (lannka) [reply src/consent.js]: SGTM

**Pull request merged**